crates/proof-of-sql/src/base/database/expr_utility.rs

```diff
@@ -0,0 +1,185 @@
+        right: Box::new(right),
+    }
+}
+
+/// Construct a new `Expr` A - B
+#[must_use]
+pub fn sub(left: Expr, right: Expr) -> Expr {
+    Expr::BinaryOp {
+        op: BinaryOperator::Minus,
+        left: Box::new(left),
+        right: Box::new(right),
+    }
+}
+
+/// Get literal from value
+pub fn lit<L>(literal: L) -> Expr
+where
+    L: Into<Literal>,
+{
+    Expr::from(literal.into())
+}
+
+/// Count the amount of non-null entries of an expression
+#[must_use]
+pub fn count(expr: Expr) -> Expr {
+    Expr::Function(Function {
+        name: ObjectName(vec![Ident::new("COUNT")]),
+        args: vec![FunctionArg::Unnamed(FunctionArgExpr::Expr(*Box::new(expr)))],
+        filter: None,
+        null_treatment: None,
+        over: None,
+        distinct: false,
+        special: false,
+        order_by: vec![],
+    })
+}
+
+/// Construct a new `Expr` representing A * B
+#[must_use]
+pub fn mul(left: Expr, right: Expr) -> Expr {
+    Expr::BinaryOp {
+        left: Box::new(left),
+        op: BinaryOperator::Multiply,
+        right: Box::new(right),
+    }
+}
+
+/// Compute the minimum of an expression
+#[must_use]
+pub fn min(expr: Expr) -> Expr {
+    Expr::Function(Function {
+        name: ObjectName(vec![Ident::new("MIN")]),
+        args: vec![FunctionArg::Unnamed(FunctionArgExpr::Expr(*Box::new(expr)))],
+        filter: None,
+        null_treatment: None,
+        over: None,
+        distinct: false,
+        special: false,
+        order_by: vec![],
+    })
+}
+
+/// Construct a new `Expr` for NOT P
+#[must_use]
+pub fn not(expr: Expr) -> Expr {
+    Expr::UnaryOp {
+        op: UnaryOperator::Not,
+        expr: Box::new(expr),
+    }
+}
+
+/// Construct a new `Expr` for A >= B
+#[must_use]
+pub fn ge(left: Expr, right: Expr) -> Expr {
+    Expr::BinaryOp {
+        left: Box::new(left),
+        op: BinaryOperator::GtEq,
+        right: Box::new(right),
+    }
+}
+
+/// Construct a new `Expr` for A == B
+#[must_use]
+pub fn equal(left: Expr, right: Expr) -> Expr {
+    Expr::BinaryOp {
+        left: Box::new(left),
+        op: BinaryOperator::Eq,
+        right: Box::new(right),
+    }
+}
+
+/// Construct a new `Expr` for P OR Q
+#[must_use]
+pub fn or(left: Expr, right: Expr) -> Expr {
+    Expr::BinaryOp {
+        left: Box::new(left),
+        op: BinaryOperator::Or,
+        right: Box::new(right),
+    }
+}
+
+// #[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+// /// An expression with an alias e.g. `a + 1 AS b`
+// pub struct TestAliasedResultExpr {
+//     /// The expression e.g. `a + 1`, `COUNT(*)`, etc.
+//     pub expr: Box<Expr>,
+//     /// The alias e.g. `count` in `COUNT(*) AS count`
+//     pub alias: Ident,
+// }
+
+// impl TestAliasedResultExpr {
+//     /// Create a new `AliasedResultExpr`
+//     #[must_use]
+//     pub fn new(expr: Expr, alias: Ident) -> Self {
+//         Self {
+//             Box::new(expr),
+//                         alias,
+//         }
+//     }
+
+// }
+
```

Comment:
Please don't included commented out code in this PR. You can put them in a different branch that depend on the branch here.